apps/src/templates/teacherDashboard/SectionActionBox.jsx

```diff
@@ -0,0 +1,216 @@
+        sectionName,
+        removeSection,
+        toggleSectionHidden,
+        importOrUpdateRoster} from './teacherSectionsRedux';
+import {sortableSectionShape, OAuthSectionTypes} from "./shapes.jsx";
+import {pegasus} from "../../lib/util/urlHelpers";
+import * as utils from '../../utils';
+import BaseDialog from '../BaseDialog';
+import Button from '../Button';
+import DialogFooter from "./DialogFooter";
+
+
+const styles = {
+  arrowIcon: {
+    paddingRight: 8
+  },
+  actionBox: {
+    width: 180,
+    paddingLeft: 15,
+    paddingRight: 15,
+    border: '1px solid ' + color.lighter_gray,
+    borderRadius: 2,
+    backgroundColor: color.white,
+    boxShadow: "2px 2px 2px " + color.light_gray,
+    marginTop: 5,
+  },
+  actionText: {
+    fontSize: 14,
+    fontFamily: '"Gotham", sans-serif',
+    color: color.charcoal,
+    padding: '10px 0px 4px 0px',
+  },
+  actionTextBreak: {
+    fontSize: 14,
+    fontFamily: '"Gotham", sans-serif',
+    borderTop: '1px solid ' + color.lighter_gray,
+    color: color.charcoal,
+    padding: '13px 0px 2px 0px',
+    marginTop: '7px',
+  },
+  archiveDelete: {
+    color: color.red,
+    padding: '10px 0px 10px 0px',
+    fontSize: 14,
+  },
+  xIcon: {
+    paddingRight: 5,
+  },
+  bold: {
+    fontFamily: '"Gotham 5r", sans-serif'
+  },
+};
+
+class SectionActionBox extends Component {
+  static propTypes = {
+    style: PropTypes.object,
+    handleEdit: PropTypes.func,
+    sectionData: sortableSectionShape.isRequired,
+
+    //Provided by redux
+    removeSection: PropTypes.func.isRequired,
+    toggleSectionHidden: PropTypes.func.isRequired,
+    sectionCode: PropTypes.string,
+    sectionName: PropTypes.string,
+    updateRoster: PropTypes.func.isRequired,
+  };
+
+  state = {
+    deleting: false,
+  };
+
+  onConfirmDelete = () => {
+      const {removeSection } = this.props;
+      const section = this.props.sectionData;
+      $.ajax({
+          url: `/v2/sections/${section.id}`,
+          method: 'DELETE',
+      }).done(() => {
+          removeSection(section.id);
+      }).fail((jqXhr, status) => {
+          // We may want to handle this more cleanly in the future, but for now this
+          // matches the experience we got in angular
+          alert(i18n.unexpectedError());
+          console.error(status);
+      });
+  };
+
+  onClickEdit = () => {
+      this.props.handleEdit(this.props.sectionData.id);
+  };
+
+  onClickHideShow = () => {
+      this.props.toggleSectionHidden(this.props.sectionData.id);
+  };
+
+  onClickSync = () => {
+    // Section code is the course ID, without the G- or C- prefix.
+    const courseId = this.props.sectionCode.replace(/^[GC]-/, '');
+    this.props.updateRoster(courseId, this.props.sectionName)
+      .then(() => {
+        // While we are embedded in an angular page, reloading is the easiest
+        // way to pick up roster changes.  Once everything is React maybe we
+        // won't need to do this.
+        utils.reload();
```

Comment:
Probably should be considered out-of-scope of this PR, but it'd be nice to address this comment and allow syncing without a page reload.  Might also want some kind of progress or loading spinner intermediate state.